exporter/otlphttpexporter/otlp.go

```diff
@@ -213,3 +218,32 @@ func readResponse(resp *http.Response) *status.Status {
 
 	return respStatus
 }
+
+type userAgentRoundTripper struct {
```

Comment:
Do we really need a roundtripper? Since this is private, we should just simply add a header to the request when creating the request instead of cloning etc., as we need to do for a proper roundtripper.

Reply:
I did consider just adding the header when request is created, which is easier but more error-prone (need to remember to add the header again, if there's ever a new codepath that creates requests). That being said, I expect the exporter is fairly stable at this point, so this should be OK? https://github.com/open-telemetry/opentelemetry-collector/pull/3970/commits/f5bddaf5d217f650fa1bbd13990e43318c15a3d2